swift-reactive-blerpc/src/main/java/com/blerpc/ios/ServiceGenerator.java

```diff
@@ -0,0 +1,224 @@
+package com.blerpc.ios;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.blerpc.proto.Blerpc;
+import com.blerpc.proto.MethodType;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.DescriptorProtos.FileDescriptorProto;
+import com.google.protobuf.DescriptorProtos.MethodDescriptorProto;
+import com.google.protobuf.DescriptorProtos.ServiceDescriptorProto;
+import com.google.protobuf.DescriptorProtos.SourceCodeInfo.Location;
+import com.google.protobuf.compiler.PluginProtos;
+import edu.umd.cs.findbugs.annotations.SuppressWarnings;
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ServiceGenerator {
+
+    private static final int METHOD_NUMBER_OF_PATHS = 4;
+    private static final int SERVICE_NUMBER_OF_PATHS = 2;
+    private static final String OUTPUT_FILE_EXTENSION = ".swift";
+    private static final String SERVICE_UUID_SEPARATOR = "-";
+    private static final String SWIFT_PACKAGE_TO_CLASS_SAPARATOR = "_";
+    private static final String METHOD_TYPE_READ = "READ";
+    private static final String METHOD_TYPE_WRITE = "WRITE";
+    private static final String METHOD_TYPE_SUBSCRIBE = "SUBSCRIBE";
+
+    public Stream<ServiceContext> buildServiceContexts(PluginProtos.CodeGeneratorRequest request) {
+        return request
+                .getProtoFileList()
+                .stream()
+                .filter(this::hasPackage)
+                .filter(file -> request.getFileToGenerateList().contains(file.getName()))
+                .flatMap(this::getFileLocations)
+                .filter(this::isProtoService)
+                .filter(this::isBleRpcService)
+                .map(fileLocation ->
+                                buildServiceContext(
+                                        fileLocation.getKey(),
+                                        fileLocation.getValue()))
+                .collect(ImmutableList.toImmutableList()).stream();
```

Comment:
import static toImmutableList.

Reply:
У меня не получилось это исправить. Когда я делаю 

```
import static com.google.common.collect.ImmutableList.toImmutableList;
```

мне потом приходит ошибка:

```
Incompatible types. Required ImmutableList<FieldContext> but 'collect' was inferred to R: no instance(s) of type variable(s) E exist so that ImmutableList<E> conforms to ImmutableList<FieldContext>
```

Reply:
Обожди, у тебя код вообще не имеет смысла: `.collect(ImmutableList.toImmutableList()).stream()` собирает поток в лист и назад конвертирует в стрим. У тебя на входе уже стрим.

Reply:
А в другом месте, если это происходит, покажи, пожалуйста, конкретный код, который ты написал.